src/main/java/com/mineboundteam/minebound/magic/MagicEvents.java

```diff
@@ -300,6 +308,49 @@ protected static NonNullConsumer<PlayerMana> handlePlayerMana(Player player) {
         };
     }
 
+    /**
+     * Whenever player changes armor or held item, post spell equip remove events
+     * @param event
+     */
+    @SubscribeEvent
+    public static void onArmorChanged(LivingEquipmentChangeEvent event){
+        if(event.getEntityLiving() instanceof ServerPlayer player){
+            if(event.getSlot().getType() == Type.HAND){
+                InteractionHand h = event.getSlot() == EquipmentSlot.MAINHAND ? InteractionHand.MAIN_HAND : InteractionHand.OFF_HAND;
+                
+                if(!event.getFrom().isEmpty() && event.getFrom().getItem() instanceof ActiveSpellItem)
+                    MinecraftForge.EVENT_BUS.post(new MagicSelectedEvent.Remove(player, event.getFrom(), new SelectedSpell(null, h, 0)));
+                if(!event.getTo().isEmpty() && event.getTo().getItem() instanceof ActiveSpellItem);
+                    MinecraftForge.EVENT_BUS.post(new MagicSelectedEvent.Select(player, event.getTo(), new SelectedSpell(null, h, 0)));
```

Comment:
There's an errant semicolon at the end of this if statement.

Reply:
Fixed